drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c

```diff
@@ -217,63 +218,107 @@ int wifi_nrf_set_power_save(const struct device *dev,
 		goto out;
 	}
 
-	status = wifi_nrf_fmac_set_power_save(rpu_ctx_zep->rpu_ctx,
-					      vif_ctx_zep->vif_idx,
-					      params->enabled);
-
-	if (status != WIFI_NRF_STATUS_SUCCESS) {
-		LOG_ERR("%s: wifi_nrf_fmac_ failed\n", __func__);
-		goto out;
-	}
-
-	ret = 0;
-out:
-	return ret;
-}
-
-
-int wifi_nrf_set_power_save_mode(const struct device *dev,
-				 struct wifi_ps_mode_params *ps_mode_params)
-{
-	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
-	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
-	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
-	int ret = -1;
-
-	if (!dev || !ps_mode_params) {
-		goto out;
-	}
-
-	vif_ctx_zep = dev->data;
-
-	if (!vif_ctx_zep) {
-		LOG_ERR("%s: vif_ctx_zep is NULL\n", __func__);
-		goto out;
-	}
-
-	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	switch (params->type) {
+	case WIFI_PS_PARAM_LISTEN_INTERVAL:
+		if ((params->listen_interval <
+		     NRF_WIFI_LISTEN_INTERVAL_MIN) ||
+		    (params->listen_interval >
+		     WIFI_LISTEN_INTERVAL_MAX)) {
+			params->fail_reason =
+				WIFI_PS_PARAM_LISTEN_INTERVAL_RANGE_INVALID;
+			return -EINVAL;
+		}
+		status = wifi_nrf_fmac_set_listen_interval(
+						rpu_ctx_zep->rpu_ctx,
+						vif_ctx_zep->vif_idx,
+						params->listen_interval);
+
+		if (status != WIFI_NRF_STATUS_SUCCESS) {
+			LOG_ERR("%s: wifi_nrf_fmac_set_listen_interval failed\n",
+				__func__);
+			params->fail_reason =
+				WIFI_PS_PARAM_FAIL_CMD_EXEC_FAIL;
+			goto out;
+		}
+	break;
+	case  WIFI_PS_PARAM_TIMEOUT:
+		if (vif_ctx_zep->if_type != NRF_WIFI_IFTYPE_STATION) {
+			LOG_ERR("%s: Operation supported only in STA mode\n",
+				__func__);
+			params->fail_reason =
+				WIFI_PS_PARAM_FAIL_CMD_EXEC_FAIL;
+			goto out;
+		}
 
-	if (!rpu_ctx_zep) {
-		LOG_ERR("%s: rpu_ctx_zep is NULL\n", __func__);
-		goto out;
-	}
+		status = wifi_nrf_fmac_set_power_save_timeout(
+							rpu_ctx_zep->rpu_ctx,
+							vif_ctx_zep->vif_idx,
+							params->timeout_ms);
+
+		if (status != WIFI_NRF_STATUS_SUCCESS) {
```

Comment:
All status checks can be moved out of switch, we can just dump `%s: Configuring %d failed, __func__, params->type`, makes code cleaner